kytos/core/auth.py

```diff
@@ -136,232 +240,113 @@ def register_core_auth_services(self):
             "auth/login/", self._authenticate_user
         )
         self.controller.api_server.register_core_endpoint(
-            "auth/users/", self._list_users
+            "auth/users/", self._list_users,
         )
         self.controller.api_server.register_core_endpoint(
-            "auth/users/<uid>", self._list_user
+            "auth/users/<username>", self._list_user
         )
         self.controller.api_server.register_core_endpoint(
             "auth/users/", self._create_user, methods=["POST"]
         )
         self.controller.api_server.register_core_endpoint(
-            "auth/users/<uid>", self._delete_user, methods=["DELETE"]
+            "auth/users/<username>", self._delete_user, methods=["DELETE"]
         )
         self.controller.api_server.register_core_endpoint(
-            "auth/users/<uid>", self._update_user, methods=["PATCH"]
+            "auth/users/<username>", self._update_user, methods=["PATCH"]
         )
 
     def _authenticate_user(self):
-        """Authenticate a user using Storehouse."""
+        """Authenticate a user using MongoDB."""
         username = request.authorization["username"]
         password = request.authorization["password"].encode()
         try:
-            user = self._find_user(username)[0].get("data")
-            if user.get("password") != hashlib.sha512(password).hexdigest():
-                raise KeyError
+            user = self._find_user(username)
+            if user["password"] != hashlib.sha512(password).hexdigest():
+                raise Unauthorized
             time_exp = datetime.datetime.utcnow() + datetime.timedelta(
                 minutes=self.token_expiration_minutes
             )
             token = self._generate_token(username, time_exp)
             return {"token": token}, HTTPStatus.OK.value
-        except (AttributeError, KeyError) as exc:
+        except (AttributeError, KeyError, Unauthorized) as exc:
```

Comment:
Aldo, since now we have a pydantic `UserDoc` model and it's being validated, all of the `KeyError` and `AttributeError` aren't expected to happen here anymore (which could happen with storehouse), if they do happen now, then it's an issue with the logic of the code (or if someone accidentally insert a bad user bypassing validations, but that shouldn't happen in the application either), and that shouldn't be considered Unauthorized and we should let it crash with a 500 error and original exception, so could you `AttributeError` and `KeyError` from he `except` clause? Let me know if you're seeing any other corner case.

Reply:
Let's also not include the `exc` in the message here, for `Unauthorized` that isn't being set, so I don't expect anything else to potentially leak there, but let's leave only the literal string `"Incorrect username or password"` to be safe.

Reply:
Ok